### ISSUE TITLE
raft topology / tablets: introduce nodetool API to mark node as permanently dead (like in `removenode` but without the actual remove)

### ISSUE THREAD

**Issue by kbr-scylla** [scylladb/scylladb#21281]
If a node is dead and cannot be brought back, tablet migrations are stuck, until the node is explicitly marked as "permanently dead" or as "ignored node" (that's how it's called in code).

Currently this is done during `removenode` and `replace` operations but it should be possible to only mark the node as dead, for the purpose of unblocking migrations, without doing the actual `removenode`, because the larger topology change might be currently impossible (see https://github.com/scylladb/scylladb/issues/21239).

cc @tgrabiec @bhalevy @gleb-cloudius

**Comment by bhalevy** [comment]
I like this idead.
I presume this action is permanent and the node would be banned from rejoining the cluster after that, and that the only allowed operation regarding it would be replace and removenode.

**Comment by kbr-scylla** [comment]
yes

**Comment by gleb-cloudius** [comment]
IIRC we changed removenode to mark a node as dead immediately (before the topology request is processed) exactly for this use case. Why do we need another API?

**Comment by kbr-scylla** [comment]
See discussion here: https://github.com/scylladb/scylladb/issues/21239#issuecomment-2431958653

**Comment by gleb-cloudius** [comment]
> See discussion here: [#21239 (comment)](https://github.com/scylladb/scylladb/issues/21239#issuecomment-2431958653)

As @tgrabiec said there even failed removenode suppose to mark a node as permanently dead.

**Comment by kbr-scylla** [comment]
> Or is the purpose of the issue is fixing a UX problem?

yes

>  Or may be add a flag to removenode to do just that

Adding flag sounds good -- `--only-mark-dead` or something

**Comment by mykaul** [comment]
I'm unsure why it's marked with 'tablets' label.

**Comment by bhalevy** [comment]
From https://github.com/scylladb/scylladb/issues/21281#issue-2614023913:
> If a node is dead and cannot be brought back, tablet migrations are stuck

**Comment by mykaul** [comment]
We probably need to give this some attention - @emdroid ?

**Comment by dani-tweig** [comment]
> We probably need to give this some attention - [@emdroid](https://github.com/emdroid) ?

@mykaul is this important for 2025.3?

**Comment by mykaul** [comment]
> > We probably need to give this some attention - [@emdroid](https://github.com/emdroid) ?
> 
> [@mykaul](https://github.com/mykaul) is this important for 2025.3?

Yes, unless there's a workaround. See the very first line of the description on why this scenario is very problematic.

**Comment by gleb-cloudius** [comment]
> > > We probably need to give this some attention - [@emdroid](https://github.com/emdroid) ?
> > 
> > 
> > [@mykaul](https://github.com/mykaul) is this important for 2025.3?
> 
> Yes, unless there's a workaround. See the very first line of the description on why this scenario is very problematic.

This is purely cosmetic issue: https://github.com/scylladb/scylladb/issues/21281#issuecomment-2439887234

**Comment by bhalevy** [comment]
As discussed today, let's consider removing excluded nodes from group0, so they won't affect the group0 quorum, if it's safe to do (it should be since the quorum is required to exclude the node).

**Comment by gleb-cloudius** [comment]
> As discussed today, let's consider removing excluded nodes from group0, so they won't affect the group0 quorum, if it's safe to do (it should be since the quorum is required to exclude the node).

Excluded nodes are suppose to be dead and as such not be voters and thus they will not affect group0 quorum. There is nothing to be done as far as I see.
